test/e2e/examples.go

```diff
@@ -157,6 +160,111 @@ var _ = Describe("Examples e2e", func() {
+			// namespaces are passed directly.
+			// Also, for simplicity, we don't use yamls with namespaces, but we
+			// create testing namespaces instead.
+
+			backendRcYaml := mkpath("dns-backend-rc.yaml")
+			backendRcName := "dns-backend"
+			backendSvcYaml := mkpath("dns-backend-service.yaml")
+			backendSvcName := "dns-backend"
+			backendPodName := "dns-backend"
+			frontendPodYaml := mkpath("dns-frontend-pod.yaml")
+			frontendPodName := "dns-frontend"
+			frontendPodContainerName := "dns-frontend"
+
+			podOutput := "Hello World!"
+
+			// we need two namespaces anyway, so let's forget about
+			// the one created in BeforeEach and create two new ones.
+			namespaces := []*api.Namespace{nil, nil}
+			for i := range namespaces {
+				var err error
+				namespaces[i], err = createTestingNS(fmt.Sprintf("dnsexample%d", i), c)
+				if namespaces[i] != nil {
+					defer c.Namespaces().Delete(namespaces[i].Name)
+				}
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			for _, ns := range namespaces {
+				runKubectl("create", "-f", backendRcYaml, getNsCmdFlag(ns))
+			}
+
+			// create two backend services
+			for _, ns := range namespaces {
+				runKubectl("create", "-f", backendSvcYaml, getNsCmdFlag(ns))
+			}
+
+			// wait for objects
+			for _, ns := range namespaces {
+				waitForRCPodsRunning(c, ns.Name, backendRcName)
+				waitForService(c, ns.Name, backendSvcName, true, poll, serviceStartTimeout)
+			}
+			// it is not enough that pods are running because they may be set to running, but
+			// the application itself may have not been initialized. Just query the application.
+			for _, ns := range namespaces {
+				label := labels.SelectorFromSet(labels.Set(map[string]string{"name": backendRcName}))
+				pods, err := c.Pods(ns.Name).List(label, fields.Everything())
+				Expect(err).NotTo(HaveOccurred())
+				err = podsResponding(c, ns.Name, backendPodName, false, pods)
+				Expect(err).NotTo(HaveOccurred())
+				Logf("found %d backend pods responding in namespace %s", len(pods.Items), ns.Name)
+
+				err = serviceResponding(c, ns.Name, backendSvcName)
+				Expect(err).NotTo(HaveOccurred(), "waiting for the service to respond")
+			}
+
+			// Now another tricky part:
+			// It may happen that the service name is not yet in DNS.
+			// So if we start our pod, it will fail. We must make sure
+			// the name is already resolvable. So let's try to query DNS from
+			// the pod we have, until we find our service name.
+			// This complicated code may be removed if the pod itself retried after
+			// dns error or timeout.
+			// This code is probably unnecessary, but let's stay on the safe side.
+			label := labels.SelectorFromSet(labels.Set(map[string]string{"name": backendPodName}))
+			pods, err := c.Pods(namespaces[0].Name).List(label, fields.Everything())
+
+			if err != nil || pods == nil || len(pods.Items) == 0 {
+				Failf("no running pods found")
+			}
+			podName := pods.Items[0].Name
+
+			pythonCode := `
```

Comment:
Ugh, this is kind of janky.

Can you move this to a const at the top of the file, and fix indentation, it will make it easier to read.